reactor-core/src/main/java/reactor/core/publisher/ChangedPredicate.java

```diff
@@ -0,0 +1,41 @@
+package reactor.core.publisher;
+
+import java.util.function.BiPredicate;
+import java.util.function.Function;
+import java.util.function.Predicate;
+
+import reactor.core.Disposable;
+
+class ChangedPredicate<T, K> implements Predicate<T>, Disposable {
```

Comment:
looks way simpler, I love it :) nitpick: in order to avoid top-level class, can you put this as a `static` package-private inner class of `FluxBufferPredicate`?